lib.php

```diff
@@ -494,54 +494,78 @@ function block_coursefeedback_get_feedbackname($feedbackid = null) {
 
     return htmlentities($name);
 }
-
 /**
+ * This function gets the amount of votes for each answeroption (1-6) and each question.
+ * It calculates the amount of counted choices, the amount of abstentions and the average rating for each question
+ *
  * @param int $courseid
  * @param string $sort
- * @return array - 2-dimensional array of answers, ordered by question id
+ * @return array - 2-dimensional array of answers, ordered by question id as follows:
+ * [ "<questionid>" =>
+ *      [ "1" => <answercount>, ..., "6" => <answercount>, "average" =>..., "choicessum" =>..., "abstentions" => ... ]]
+ * @throws moodle_exception
  */
-function block_coursefeedback_get_answers($course, $feedbackid, $sort = "questionid") {
-    global $DB, $CFG, $USER;
+function block_coursefeedback_get_qanswercounts($course, $feedbackid) {
+    global $DB;
     $config = get_config("block_coursefeedback");
     $answers = array();
     $course = clean_param($course, PARAM_INT);
 
     if ($course <= 0) {
         throw new moodle_exception("invalidcourseid");
     }
-
-    $questions = block_coursefeedback_get_questions($feedbackid, $config->default_language);
+    // Get all the questions of the feedback
+    $questions = block_coursefeedback_get_questions_by_language($feedbackid, [current_language()]);
     $params = array("fid" => $feedbackid, "course" => $course);
     if (!empty($questions)) {
-        $count = count($questions);
-        foreach (array_keys($questions) as $question) {
-            $params["qid"] = $question;
+        // For each question and each answerpossibility, count the amount of the given answers
+        foreach ($questions as $question) {
+            $choicessum = 0;
+            $avsum = 0;
+            $questionid = $question->questionid;
+            $params["qid"] = $questionid;
             $sql = "SELECT answer,COUNT(*) AS count
-                      FROM {block_coursefeedback_answers}
-                     WHERE coursefeedbackid = :fid 
-                           AND questionid = :qid 
-                           AND course = :course
-                  GROUP BY answer";
-
+                    FROM {block_coursefeedback_answers}
+                    WHERE coursefeedbackid = :fid 
+                        AND questionid = :qid 
+                        AND course = :course
+                    GROUP BY answer";
+
+            // Create array for the question and fill in zeros for each answeroption
+            $answers[$questionid] = array_fill(1, 6, 0);
+
+            $answers[$questionid]['average'] = "x";
```

Comment:
Wieso setzt du hier `"x"` als Mittelwert? Das ist ja schon anders als vorher, für den Fall, dass es keine Antworten zu der Frage gab, oder? Dieser wird ja in dem Fall nie mit `0` überschrieben.

Reply:
Ja, die Idee ist, dass es ja keinen Mittelwert gibt z.B. wenn es nur 2 Enthaltungen gegeben hätte. Ich finde "x" drückt dann besser aus, dass es eben keinen Mittelwert gibt. Sobald es antworten auf die Frage gibt, wird der Mittelwert berechnet und das "x" überschrieben.

Reply:
Ich schon. X ist kein Mittelwert und wenn später irgendwer dran weiterarbeitet und das nicht weiß könnte da einiges an seltsamen Verhalten auftreten. Das Günstigste wäre, dass "x" einfach zur 0 gecastet wird, das Schlimmste wenn der ASCII-Wert verwendet wird. 
Aber auch wenn es textuell einfach angezeigt wird wundert man sich woher da ein x kommt. Ich sehe da viele andere Möglichkeiten, das sinnvoller zu gestalten. Du kannst zum einen FALSE/NULL setzen, wenn es denn unbedingt gebraucht wird, damit es sicher genullt wird, wenn man damit rechnet und es einen leeren String bei der Ausgabe gibt. Du kannst das Feld einfach erst einmal nicht setzen und dann immer gucken, ob es das gibt (dann weißt du auch, ob die Rückgabe deiner Funktion sinnvoll war). Oder du gibst generell einfach nur FALSE zurück, wenn was schief geht anstelle eines kaputten, nicht zu gebrauchenen Arrays.
Suche dir was daraus aus... aber nicht "x".

Reply:
Resolved in 38d8e093359ee6538a55817e1fa9d81af33f32c3